TowerForge/TowerForge/GameModule/Systems/GameSystems/HealthSystem.swift

```diff
@@ -27,8 +27,18 @@ class HealthSystem: TFSystem {
 
         healthComponent.adjustHealth(amount: hp)
 
-        if healthComponent.currentHealth <= 0 {
+        guard healthComponent.currentHealth <= 0 else {
+            return
+        }
+
+        guard let currentPlayer = eventManager.currentPlayer else {
             eventManager.add(KillEvent(on: currentEntity.id, at: CACurrentMediaTime(), player: playerComponent.player))
```

Comment:
so if the `eventManager` is the host and the `currentPlayer` at the same time, does it have to go to the next block too?

Reply:
The code only executes if the player is the host. This so that the remote event is only sent once to the firebase which both players subscribe to.